### PR TITLE
Handle command outcome subscriptions

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR improves the way command outcome subscriptions are handled.

### CODE REVIEW

File: src/main/java/io/spine/helloworld/client/Client.java [hunk 1]
```diff
@@ -88,6 +89,24 @@ private void printEvent(EventMessage e) {
         System.out.println(out);
     }
 
+    /**
+     * Cancels all current subscriptions, if any.
+     */
+    private void cancelSubscriptions() {
+        if (subscriptions != null) {
+            subscriptions.forEach(s -> client.subscriptions().cancel(s));
```
Comment (yuri-sergiichuk) [inline src/main/java/io/spine/helloworld/client/Client.java:97]: Can we extract the `client.subscriptions()` call result into a variable and then use a method reference here?

```
Subscriptions subscriptions = client.subscriptions();
this.subscriptions.forEach(subscriptions::cancel);
```

File: src/main/java/io/spine/helloworld/client/Client.java [hunk 2]
```diff
@@ -88,6 +89,24 @@ private void printEvent(EventMessage e) {
...
+            this.subscriptions = null;
+        }
+    }
+
+    /**
+     * Verifies if the client finished cancelling active subscriptions.
```
Comment (yuri-sergiichuk) [inline src/main/java/io/spine/helloworld/client/Client.java:103]: Maybe not `Verifies` but `tests` or `examines`? We don't perform any active actions, right? I may be mixing up contexts, but usually `verify` does something and not just returns the state.

File: src/main/java/io/spine/helloworld/client/Client.java (hunk 1 above)
Comment (alexander-yevsyukov) [reply src/main/java/io/spine/helloworld/client/Client.java:97]: I've tried but it looks a bit like a programming puzzle: Which subscriptions cancel which?

```java
    private void cancelSubscriptions() {
        if (subscriptions != null) {
            Subscriptions subscriptions = client.subscriptions();
            this.subscriptions.forEach(subscriptions::cancel);
            this.subscriptions = null;
        }
    }
```

File: src/main/java/io/spine/helloworld/client/Client.java (hunk 2 above)
Comment (alexander-yevsyukov) [reply src/main/java/io/spine/helloworld/client/Client.java:103]: Right.

**Pull request merged**